conda/gateways/connection/session.py

```diff
@@ -218,6 +245,10 @@ def __init__(self, auth: AuthBase | tuple[str, str] | None = None):
 
         self.headers["User-Agent"] = context.user_agent
 
+        if plugin_headers is not None:
+            for header_name, header_value in plugin_headers.items():
+                self.headers[header_name] = header_value
```

Comment:
```suggestion
        if request_headers is not None:
            self.headers.update(request_headers)
```

Reply:
I'm curious why codecov doesn't detect test coverage for this though, can you verify this please?

Reply:
This was an area I overlooked. Will add tests for it.